Mage/Command/BuiltIn/CompileCommand.php

```diff
@@ -22,19 +22,43 @@
 class CompileCommand extends AbstractCommand
 {
     /**
+     * @var Compiler
+     */
+    private $compiler;
+
+    public function __construct()
+    {
+        $this->setCompiler(new Compiler());
+    }
+
+    public function setCompiler(Compiler $compiler)
```

Comment:
API is ready but it still need improvements. It should be updated per deploy. I think it's subject for another issue :)
I don't know how other IDEs behave, but in PHPStorm it's enough to give static-type of method's param. If it's what you're talking about.